wlauto/common/android/workload.py

```diff
@@ -494,6 +524,10 @@ def reset(self, context):
     def run(self, context):
         ReventWorkload.run(self, context)
 
+        # state detection check if it's enabled in the config
+        if self.check_states:
```

Comment:
I would move this check to the teardown because this would effect what instrumentation measures for the apps "run"